examples/python3/wt_session.py

```diff
@@ -0,0 +1,534 @@
+This module defines a class that manages an authenticated session
+with the WikiTree API.
+
+It allows you to log in and perform various queries.
+
+It requires python3 and the requests library.
+"""
+
+# Standard Imports
+from getpass import getpass
+import re
+from pprint import pprint
+from typing import Optional, Union
+
+# Third party imports
+import requests
+
+
+# Define the WikiTree API endpoint
+API_URL = "https://api.wikitree.com/api.php"
+
+
+class WTSession:
+    """
+    A class to manage an authenticated session on the WikiTree API.
+
+    Provides convenience functions for each of the API calls
+    """
+
+    def __init__(self) -> None:
+        self._email = ""
+        self._authenticated = False
+        self._api_session = None
+        self._authcode = None
+        self._wt_cookie = None
+        self._user_name = ""
+        self._user_id = ""
+
+    @property
+    def user_name(self) -> str:
+        """Return the user name of the authenticated user."""
+        return self._user_name
+
+    @property
+    def user_id(self) -> int:
+        """Return the user ID of the authenticated user."""
+        return self._user_id
+
+    @property
+    def email(self) -> str:
+        """Return the email of the authenticated user."""
+        return self._email
+
+    def authenticate(self, email: str, password: str) -> bool:
+        """
+        Takes an email address and password and attempts to authenticate. Returns
+        a boolean to indicate success.
+
+        :param email: The email address of the user.
+        :param password: The corresponding password.
+        :returns: Boolean indicating success.
+        """
+
+        # Deliberately not storing password as instance variable
+        self._email = email
+
+        # We use a Session to hold the state (via cookie jar) for the API queries
+        print("Starting Session")
+        self._api_session = requests.Session()
+
+        # Step 1 - POST the clientLogin action with our member credentials.
+        #
+        # We post with allow_redirect = False because we don't want the
+        # Python request to automatically follow redirects. We want the
+        # initial response (still on api.wikitree.com)
+        # and not the redirected-to destination.
+        #
+        # Note that the api_response here will, on successful login, be a
+        # redirection. In a browser interaction we'd include a returnURL
+        # value in post_data to send the browser back to our app. Here
+        # we're just capturing the authcode from the Location redirection
+        # directly.
+
+        post_data = {
+            "action": "clientLogin",
+            "doLogin": 1,
+            "wpEmail": self._email,
+            "wpPassword": password,
+        }
+        # print("POSTing clientLogin...")
+        api_response = self._api_session.post(
+            API_URL,
+            data=post_data,
+            allow_redirects=False,
+            auth=("wikitree", "wikitree"),
```

Comment:
`auth=("wikitree", "wikitree")`
Is this necessary? I've tried it without this and it works as well.

Reply:
I am unsure - have commented them out rather than deleted them.